File: StrategyPattern/bank/ConcreteWithdraw1.java
package bank;

public class ConcreteWithdraw1 implements WithdrawStrategy
{
	public void withdraw(Account acc, int money)
	{
		int balance = acc.getBalance();
		if(balance >= money)
		{
			balance -= money;
			acc.setBalance(balance);
			System.out.println("Withdraw Money : " + money + ", Balance : " + balance);
		}
		else
		{
			System.out.println("Withdraw Money is more than balance!");
		}
	}
}

File: StrategyPattern_C++/ConcreteWithdraw1.cpp
#include "ConcreteWithdraw1.h"
#include "Account.h"


ConcreteWithdraw1::ConcreteWithdraw1()
{
}


ConcreteWithdraw1::~ConcreteWithdraw1()
{
}

int ConcreteWithdraw1::withdraw(Account* acc, int money)
{
	int balance = acc->getBalance();

	if (balance >= money)
	{
		balance -= money;
		cout << "Withdraw Money : " << money << ", Balance : " << balance << endl;
	}
	else
	{
		cout << "Withdraw Money is more than balance!" << endl;
	}

	return balance;
}
